File: contests/abs/abc083_b.hs
import Data.Functor ((<&>))
import Data.Ix (inRange)
import Flow

digits :: Int -> [Int]
digits 0 = []
digits n = m : digits d
  where
    (d, m) = n `divMod` 10

main :: IO ()
main = do
  [n, a, b] <- getLine <&> words .> map (read @Int)
  [1 .. n] |> filter (digits .> sum .> inRange (a, b)) |> sum |> print

File: contests/abs/abc083_b.fs
let rec digits n =
    let struct (d, r) = System.Math.DivRem(n, 10UL)
    if n = 0UL then [] else r :: digits d

let inRange (l, r) n = l <= n && n <= r

let [| n; a; b |] = stdin.ReadLine().Split() |> Array.map uint64

[| 1UL .. n |]
|> Array.filter (digits >> List.sum >> inRange (a, b))
|> Array.sum
|> printfn "%d"
